lib/queuePopulator/IngestionProducer.js

```diff
@@ -95,16 +147,39 @@ class IngestionProducer {
      * generate a listing of all current objects that exists on the source
      * bucket, including the bucket MD to create the correct entries
      * @param {string} bucketName - name of source bucket
-     * @param {function} done - callback function
-     * @return {object} resLog value
-     *
+     * @param {object} state - previous state used to paginate version listing
+     * @param {string} [state.versionMarker] - NextVersionIdMarker
+     * @param {string} [state.keyMarker] - NextKeyMarker
+     * @param {function} done - callback(error, response) where response has:
+     *   logRes {object} - metadata logs formed as RaftLogEntry put entries
+     *   initState {object} - returns status for snapshot process
+     *   initState.status {string} - STATUS_COMPLETE || STATUS_INCOMPLETE
+     *   [initState.versionMarker] {string} - NextVersionIdMarker, if any
+     *   [initState.keyMarker] {string} - KeyMarker, if any
+     * @return {undefined}
      */
-    snapshot(bucketName, done) {
+    snapshot(bucketName, state, done) {
         async.waterfall([
-            next => this._getBucketObjects([bucketName], next),
-            (bucketList, next) =>
-                this._getBucketObjectsMetadata(bucketList, next),
-        ], err => done(err, this.resLog));
+            next => this._getObjectVersionsList(bucketName, state, next),
+            (data, next) => {
+                const {
+                    IsTruncated, versionList, versionMarker, keyMarker,
+                } = data;
+                this._getBucketObjectsMetadata(bucketName, versionList,
+                (err, logRes) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    const status = IsTruncated ?
```

Comment:
Seems like the global vars are unneeded. Could it be simplified to just be a boolean? For example:

```
initState: {
    isStatusComplete: <bool>,
    [...]
}
```

That way you can just set it to whatever the value of `IsTruncated` is.

Reply:
I like this idea 👍 I'll look into a rf